application/src/main/java/org/togetherjava/tjbot/commands/moderation/TemporaryModerationRoutine.java

```diff
@@ -0,0 +1,114 @@
+package org.togetherjava.tjbot.commands.moderation;
+
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.exceptions.ErrorResponseException;
+import net.dv8tion.jda.api.requests.ErrorResponse;
+import org.jetbrains.annotations.NotNull;
+import org.jooq.Result;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.togetherjava.tjbot.db.Database;
+import org.togetherjava.tjbot.db.generated.tables.TemporaryModerationActions;
+import org.togetherjava.tjbot.db.generated.tables.records.TemporaryModerationActionsRecord;
+
+import java.time.Instant;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+public final class TemporaryModerationRoutine {
+    private static final Logger logger = LoggerFactory.getLogger(TemporaryModerationRoutine.class);
+
+    private final Database database;
+    private final JDA jda;
+    private final ScheduledExecutorService checkExpiredActionsService =
+            Executors.newSingleThreadScheduledExecutor();
+
+    /**
+     * Creates a new instance.
+     *
+     * @param jda the JDA instance to use to send messages and retrieve information
+     * @param database the database used to retrieve data about temporary moderation actions
+     */
+    public TemporaryModerationRoutine(@NotNull JDA jda, @NotNull Database database) {
+        this.database = database;
+        this.jda = jda;
+    }
+
+    private void checkExpiredActions() {
+        logger.debug("Checking expired temporary moderation actions to revoke...");
+
+        // TODO Overlapping actions
+
+        database.write(context -> {
+            Result<TemporaryModerationActionsRecord> oldRecords = context
+                .selectFrom(TemporaryModerationActions.TEMPORARY_MODERATION_ACTIONS)
+                .where(TemporaryModerationActions.TEMPORARY_MODERATION_ACTIONS.ACTION_EXPIRES
+                    .lessOrEqual(Instant.now()))
+                .fetch();
+
+            oldRecords.forEach(recordToRevoke -> {
+                revokeAction(recordToRevoke.getUserId(), recordToRevoke.getGuildId(),
+                        ModerationUtils.Action.valueOf(recordToRevoke.getActionType()));
+                recordToRevoke.delete();
+            });
+        });
+
+        logger.debug("Finished checking expired temporary moderation actions to revoke.");
+    }
+
+    private void revokeAction(long userId, long guildId, @NotNull ModerationUtils.Action action) {
+        Guild guild = jda.getGuildById(guildId);
+        if (guild == null) {
+            logger.info(
+                    "Attempted to revoke a temporary moderation action but the bot is not connected to the guild '{}' anymore, skipping revoking.",
+                    guildId);
+            return;
+        }
+        if (action != ModerationUtils.Action.BAN) {
```

Comment:
You could call this before retrieving the guild on line 62 `jda.getGuildById(guildId);`
If the action type is unsupported, it doesn't matter of the bot is connected or not to the guild, and we won't make an "unnecessary" call to the getGuildById